api/db.py

```diff
@@ -85,37 +92,77 @@ def get_data_list(start_query, limit, offset, order_by, qs):
 
 # create query that returns data list
 def create_query_list(start_query, limit: int = 0, offset: int = 0, order_by=None, qs: str | None = None):
+    # options: limit, offset, order_by
     where_optional_args = []
     # options: limit, offset, order_by
     other_optional_args = []
     # values to put in placeholders in query
     values = []
-
-    used_clause = ["limit", "offset", "order_by"]
-
+    crc32c=False
     # add arguments to where clause
-    if qs is not None:
+    if qs:
         qs = qs.split('&')
         for item in qs:
+            # dealing with crc32c format
+            if item[-2:]=='==':
+                item = item[:-2]
+                crc32c=True
             key, val = item.split('=')
-            where_optional_args.append('"{0}" = %s'.format(key))
-            values.append(val)
-    # for key, val in request.args.items():
-    #     if key not in used_clause:
-    #         where_optional_args.append('"{0}" = %s'.format(key))
-    #         values.append(val)
+            # checking if it's crc32c format
+            if crc32c:
+                val+='=='
+                crc32c=False
+            # checks if it's simple array of strings or integers
+            if '[' in val:
+                val_splited=val[1:-1].split(',')
+                # checks if it's integers array
+                try:
+                    val = [int(element) for element in val_splited]
+                    is_all_digits = all(str(element).isdigit() for element in val)
```

Comment:
I didn't quite understand what do you mean, can you give an example/clarify?
I want to check here if each element is a number: `val = [int(element) for element in val_splited]`
if some element is not int,` int(element)` will fail it will go to except where it will preform the quotes replacement on the string array. There can be more methods to deal with this but I think this method doing the job. `is_all_digits = all(str(element).isdigit() for element in val)` not needed here I think, I don't remember why I put it, it works without it.